spec/models/accounting/quickbooks/connector_spec.rb

```diff
@@ -0,0 +1,99 @@
+  let(:invalid_realm_id) { '' }
+
+  shared_examples_for 'no token is present' do
+    it 'should not be connected' do
+      expect(connector.connected?).to be_falsey
+    end
+    it 'should not be renewable' do
+      expect(connector.renewable?).to be_falsey
+    end
+    it 'should not be expired' do
+      expect(connector.expired?).to be_falsey
+    end
+  end
+
+  context 'with new token' do
+    subject(:connector) do
+      Accounting::Quickbooks::Connector.new(quickbooks: { token: valid_token, secret: valid_secret, realm_id: valid_realm_id, token_expires_at: 180.days.from_now.utc })
+    end
+
+    it 'should be connected' do
+      expect(connector.connected?).to be_truthy
+    end
+    it 'should not be renewable' do
+      expect(connector.renewable?).to be_falsey
+    end
+    it 'should not be expired' do
+      expect(connector.expired?).to be_falsey
+    end
+  end
+
+  context 'with expiring token' do
+    subject(:connector) do
+      Accounting::Quickbooks::Connector.new(quickbooks: { token: valid_token, secret: valid_secret, realm_id: valid_realm_id, token_expires_at: 20.days.from_now.utc })
+    end
+
+    it 'should be connected' do
+      expect(connector.connected?).to be_truthy
+    end
+    it 'should be renewable' do
+      expect(connector.renewable?).to be_truthy
+    end
+    it 'should not be expired' do
+      expect(connector.expired?).to be_falsey
+    end
+  end
+
+  context 'with expired token' do
+    subject(:connector) do
+      Accounting::Quickbooks::Connector.new(quickbooks: { token: valid_token, secret: valid_secret, realm_id: valid_realm_id, token_expires_at: 2.days.ago.utc })
+    end
+    it 'should not be connected' do
+      expect(connector.connected?).to be_falsey
+    end
+    it 'should not be renewable' do
+      expect(connector.renewable?).to be_falsey
+    end
+    it 'should be expired' do
+      expect(connector.expired?).to be_truthy
+    end
+  end
+
+  context 'without token' do
+    subject(:connector) { Accounting::Quickbooks::Connector.new }
+
+    it_behaves_like 'no token is present'
+  end
+
+  context 'with invalid token' do
+    subject(:connector) do
+      Accounting::Quickbooks::Connector.new(quickbooks: { token: invalid_token, secret: valid_secret, realm_id: valid_realm_id, token_expires_at: 180.days.from_now.utc })
+    end
+
+    it_behaves_like 'no token is present'
+  end
+
+  context 'with invalid secret' do
+    subject(:connector) do
+      Accounting::Quickbooks::Connector.new(quickbooks: { token: valid_token, secret: invalid_secret, realm_id: valid_realm_id, token_expires_at: 180.days.from_now.utc })
+    end
+
+    it_behaves_like 'no token is present'
+  end
+
+  context 'with invalid realm id' do
+    subject(:connector) do
+      Accounting::Quickbooks::Connector.new(quickbooks: { token: valid_token, secret: valid_secret, realm_id: invalid_realm_id, token_expires_at: 180.days.from_now.utc })
+    end
+
+    it_behaves_like 'no token is present'
+  end
+end
```

Comment:
There should be a feature spec for the new settings page right? Testing connection will be hard and I don't think we should go down that road now unless it's easier than I'm thinking, but you could add a fake token and test disconnection, right?